it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,362 @@
+                .newBuilder()
+                .setDestinationConnectionProfile(ConnectionProfileName
+                        .format(projectId, location, connectionProfileId));
+
+        BigQueryDestinationConfig.Builder bigQueryDestinationConfig = BigQueryDestinationConfig.newBuilder()
+                .setDataFreshness(stalenessLimit)
+                .setSingleTargetDataset(BigQueryDestinationConfig
+                        .SingleTargetDataset.newBuilder().setDatasetId(datasetId));
+
+        destinationConfigBuilder.setBigqueryDestinationConfig(bigQueryDestinationConfig);
+        return destinationConfigBuilder.build();
+    }
+
+    @Override
+    public Stream createStream(String streamId,
+                               SourceConfig sourceConfig,
+                               DestinationConfig destinationConfig) {
+
+        LOG.info("Creating Stream {} in project {}.", streamId, projectId);
+
+        try {
+            CreateStreamRequest request =
+                    CreateStreamRequest.newBuilder()
+                            .setParent(LocationName.format(projectId, location))
+                            .setStreamId(streamId)
+                            .setStream(Stream
+                                    .newBuilder()
+                                    .setSourceConfig(sourceConfig)
+                                    .setDisplayName(streamId)
+                                    .setDestinationConfig(destinationConfig)
+                                    .setBackfillAll(Stream.BackfillAllStrategy.getDefaultInstance())
+                                    .build())
+                            .build();
+
+            Stream reference = datastreamClient.createStreamAsync(request).get();
+            createdStreamIds.add(streamId);
+
+            return reference;
+        } catch (ExecutionException | InterruptedException e) {
+            throw new DatastreamResourceManagerException("Failed to create stream. ", e);
+        }
+    }
+
+    public Stream updateStreamState(String streamId, Stream.State state) {
+
+        LOG.info("Updating {}'s state to {} in project {}.", streamId, state.name(), projectId);
+
+        try {
+            Stream.Builder streamBuilder = Stream.newBuilder()
+                    .setName(StreamName.format(projectId, location, streamId))
+                    .setState(state);
+
+            FieldMask.Builder fieldMaskBuilder = FieldMask.newBuilder().addPaths(FIELD_STATE);
+
+            UpdateStreamRequest request = UpdateStreamRequest.newBuilder()
+                    .setStream(streamBuilder)
+                    .setUpdateMask(fieldMaskBuilder)
+                    .build();
+
+            return datastreamClient.updateStreamAsync(request).get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatastreamResourceManagerException("Failed to update stream. ", e);
+        }
+    }
```

Comment:
I would add a log at the end that specifies the state has been successfully updated, and to which state it was updated to.